cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -274,6 +275,11 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
           twillPreparer.setSchedulerQueue(schedulerQueueName);
         }
 
+        // Add secrets if using secrets
+        if (twillPreparer instanceof SecureTwillPreparer) {
+          twillPreparer = ((SecureTwillPreparer) twillPreparer).withExternalSecret("cdap-security");
```

Comment:
Where does the name `cdap-security` comes from?

Reply:
It's actually passed via the `SecuritySecret` parameter in the cdap-cr in cdap-operator, but has never been used in CDAP itself until now. I can go ahead and make it configurable in cConf.